common/util.go

```diff
@@ -238,7 +238,9 @@ func ValidIDLength(
 	idLength := len(id)
 	valid := idLength <= errorLimit
 	if idLength > warnLimit {
-		scope.IncCounter(metricsCounter)
+		if scope != nil {
+			scope.IncCounter(metricsCounter)
```

Comment:
use `tally.Noop` instead. We typically assume scope and logger are not nil.

Reply:
Hmm... But we had the check for `logger`, so I assumed that we should have the same for `scope`. 
It's not obvious how to call `tally.Noop`, because [`NoopScope`](https://github.com/uber/cadence/blob/master/common/metrics/scope.go#L51) requires some `ServiceIdx`.

Reply:
True, in this case it's more coherent if `scope != nil` check is present.
But it's super unusual - we almost never do that neither for logs nor for metrics.
I meant  just `tally.NoopScope`.

Reply:
Maybe it is worth doing in a separate PR, but I think it could be done here.

Reply:
@3vilhamster I have created another [PR](https://github.com/uber/cadence/pull/5526) with your suggestions 👍